### DISCUSSION TITLE
[General] <create contract on zksync Sepolia failed with foundry>

### DISCUSSION THREAD

**Discussion by wiselyG** [zkSync-Community-Hub/zksync-developers#829] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Im use foundry to build contract on zksync Sepolia,this is command,

```
forge script script/Counter.s.sol --rpc-url $ZKSYNC_TESTNET --account zksyncSep  --broadcast -vvv
```
but i get failed,the error is:
```
Error: Failed to send transaction

Context:
- server returned an error response: error code 3: Failed to serialize transaction: toAddressIsNull
```
My forge version:forge 0.2.0 (00efa0d5 2024-12-07T15:44:15.859516877Z)
This is the demo contract
```
// SPDX-License-Identifier: UNLICENSED
pragma solidity 0.8.18;

contract Counter {
    uint256 public number;

    function setNumber(uint256 newNumber) public {
        number = newNumber;
    }

    function increment() public {
        number++;
    }
}
```
This is the deploy script:
```
// SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.13;

import {Script, console} from "forge-std/Script.sol";
import {Counter} from "../src/Counter.sol";

contract CounterScript is Script {
    Counter public counter;

    function setUp() public {}

    function run() public {
        vm.startBroadcast();

        counter = new Counter();

        vm.stopBroadcast();
    }
}
```

**Comment by alhenaai** [comment]
The error you're encountering, "Failed to serialize transaction: toAddressIsNull," typically indicates that the transaction is missing a necessary address parameter, such as the contract address for deployment. This can happen if the deployment script does not properly specify the address or if there's an issue with how the script interacts with the network.

Here are a few steps you can take to troubleshoot and resolve this issue:

1. **Check the RPC URL**: Ensure that the `$ZKSYNC_TESTNET` environment variable is correctly set to the RPC URL of the zkSync Sepolia testnet. This should be a complete and correct URL.

2. **Verify Contract Deployment**: Make sure that the contract `Counter` is correctly compiled and the artifact is available for deployment. Sometimes, issues can arise if the contract compilation is not up to date or if the artifact is missing.

3. **Script Review**: Review your deployment script to ensure that all necessary parameters are correctly set, especially those related to network configurations and contract interactions.

4. **Forge Version**: You mentioned using Forge version 0.2.0. Ensure that this version is compatible with the zkSync Sepolia testnet and that there are no known issues with this version related to deployment on zkSync.

5. **Update and Retry**: If possible, update your Foundry tools to the latest version and retry the deployment. Sometimes, bugs fixed in newer versions can resolve these issues.

If the problem persists after trying these steps, consider seeking further assistance by posting a detailed question with the error logs and setup details on the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions). This platform can provide more in-depth support from other developers who might have faced similar issues.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by dutterbutter** [comment, marked as answer]
Hey you have to include `--zksync` when making use of [foundry-zksync](https://github.com/matter-labs/foundry-zksync). The command would then be:

`forge script script/Counter.s.sol --rpc-url $ZKSYNC_TESTNET --account zksyncSep  --broadcast -vvv --zksync`
